smartsim/ml/data.py

```diff
@@ -235,195 +223,121 @@ class StaticDataDownloader:
-        """Initalize list of data sources based on incoming entitites and self.sub_indices.
-
-
-        Each source is represented as a tuple `(producer_name, sub_index)`.
-        Before `init_samples()` is called, the user can modify the list.
-        Once `init_samples()` is called, all data is downloaded and batches
-        can be obtained with iter(). The list of all sources is stored as `self.sources`.
-
-        :raises ValueError: If self.uploader_info is set to `auto` but no `uploader_name` is specified.
-        :raises ValueError: If self.uploader_info is not set to `auto` or `manual`.
-        """
-        self.client = Client(self.address, self.cluster)
-        if self.uploader_info == "auto":
-            if not self.uploader_name:
-                raise ValueError(
-                    "uploader_name can not be empty if uploader_info is 'auto'"
-                )
-            self._get_uploader_info(self.uploader_name)
-        elif self.uploader_info == "manual":
-            pass
-        else:
-            raise ValueError(
-                f"uploader_info must be one of 'auto' or 'manual', but was {self.uploader_info}"
-            )
-
-        self.sources = self._list_all_sources()
-
     @property
     def need_targets(self):
         """Compute if targets have to be downloaded.
 
         :return: Whether targets (or labels) should be downloaded
         :rtype: bool
         """
-        return self.target_prefix and not self.autoencoding
+        return self.target_name and not self.autoencoding
 
     def __len__(self):
         length = int(np.floor(self.num_samples / self.batch_size))
         return length
 
     def __iter__(self):
 
-        if self.sources:
-            self.update_data()
-            # Generate data
-            if len(self) < 1:
-                msg = "Not enough samples in generator for one batch. "
-                msg += "Please run init_samples() or initialize generator with init_samples=True"
-                raise ValueError(msg)
+        self.update_data()
+        # Generate data
+        if len(self) < 1:
+            msg = "Not enough samples in generator for one batch. "
+            msg += "Please run init_samples() or initialize generator with init_samples=True"
+            raise ValueError(msg)
 
-            for index in range(len(self)):
-                indices = self.indices[
-                    index * self.batch_size : (index + 1) * self.batch_size
-                ]
+        for index in range(len(self)):
+            indices = self.indices[
+                index * self.batch_size : (index + 1) * self.batch_size
+            ]
 
-                x, y = self.__data_generation(indices)
+            x, y = self.__data_generation(indices)
 
-                if y is not None:
-                    yield x, y
-                else:
-                    yield x
+            if y is not None:
+                yield x, y
+            else:
+                yield x
```

Comment:
Per our conversation: It would require the user to have to manually unpack the iterated tuples via:
```py
for val_i_care_about, know_this_to_be_none in autoencoder:
    assert know_this_to_be_none is None  # always true
    ...
```
Let's not force this on our users. They'll have a better understanding of how their `DataDownloader`s supposed to work and should know the type of objects they are iterating over.

As long as the return type does not change during iteration, via something like:
```py
data_iter = iter(DataDownloader(...))
val_1 = next(data_iter)
assert isinstance(val_1, tuple)
val_2 = next(data_iter)
assert not isinstance(val_2, tuple)
```
I think the current implementation is fine!!

Feel free to do the syntax update if you'd like, but that's all extra credit ;)